google/cloud/forseti/services/inventory/storage.py

```diff
@@ -730,13 +713,26 @@ def write(self, resource):
         Raises:
             Exception: If storage was opened readonly.
         """
-
         if self.readonly:
             raise Exception('Opened storage readonly')
 
+        previous_id = self._get_resource_id(resource)
+        if previous_id:
+            resource.set_inventory_key(previous_id)
+            self.update(resource)
+            return
+
         rows = Inventory.from_resource(self.inventory_index, resource)
+
+        # Insert the resource row to get the inventory id, buffer other rows.
         for row in rows:
-            self.buffer.add(row)
+            if row.category == Categories.resource:
+                self.session.add(row)
+                self.session.flush()
+                resource.set_inventory_key(row.id)
+            else:
+                row.parent_id = resource.inventory_key()
+                self.buffer.add(row)
```

Comment:
self.session.flush will not flush the buffer, you will need to call the commit() method in the Storage class to do that.

Reply:
commit is still done like before when the inventory is complete, nothing changed there.